File: client/src/components/Product.js
import React, { useEffect, useState } from "react";
import {Card, CardBody} from "reactstrap";

export default function Product({ product }) {
  return (
    <Card>
      <CardBody>
        <h3>{product.name}</h3>
        <p>{product.productDetail.description}</p>
        <p>{product.price}</p>
        <p>{product.quantity}</p>
      </CardBody>
    </Card>
  );
}

File: PigAndBaconTannery/Models/Product.cs
using System.Collections.Generic;

namespace PigAndBaconTannery.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public int VendorId { get; set; }
        public Vendor Vendor { get; set; }
        public ProductDetail ProductDetail { get; set; }
        public List<int> CategoryIds { get; set; }


    }
}